# Python Task 02: Operators

#### Task Overview
Welcome to your Python Operators Task!
In this task, you'll dive deep into one of the core building blocks of Python called Operators. You'll practice working with different types of operators including Arithmetic, Logical, and Comparison operators. These operators are essential for performing calculations, making decisions, and building real-world Python programs.
By the end of this task, you’ll be able to:

- Use arithmetic operators to perform mathematical operations.


- Apply logical operators to combine multiple conditions.


- Compare values using comparison operators.


This task will strengthen your foundational knowledge of Python and prepare you for building more complex logic in your future coding projects.

---------------------------------------------------------------------------------------------------------------------------------------------


# Operators

### 1. Arithmetic Operators

#### Questions:

#### 1. What are arithmetic operators? Give 2 examples in your own words.

Arithmetic operators are symbols used in programming to perform basic mathematical operations such as addition, subtraction, multiplication, division, etc.

##### Examples in my own words:

1. Addition (+) → If I write 10 + 5, the operator + adds the two numbers and gives result 15.

2. Multiplication (*) → If I write 6 * 3, the operator * multiplies the numbers and gives result 18.

#### 2. Define below two variables and perform operations using arithmetic operators (+, -, *, /, //, %, **).
value1 = 10
value2 = 15

- Add value1 and value2.
- Subtract value1 from value2.
- Multiply value1 and value2.
- Divide value1 by value2.
- Find the floor division of value1 by value2.
- Find the remainder when dividing value1 by value2.
- Calculate value1 raised to the power of value2.


In [24]:
# Defining variables
value1 = 10
value2 = 15

# Operations
print("Addition:", value1 + value2)          # 10 + 15 = 25
print("Subtraction:", value2 - value1)      # 15 - 10 = 5
print("Multiplication:", value1 * value2)   # 10 * 15 = 150
print("Division:", value1 / value2)         # 10 / 15 = 0.666...
print("Floor Division:", value1 // value2)  # 10 // 15 = 0
print("Remainder:", value1 % value2)        # 10 % 15 = 10
print("Power:", value1 ** value2)           # 10 ** 15 = 100...

Addition: 25
Subtraction: 5
Multiplication: 150
Division: 0.6666666666666666
Floor Division: 0
Remainder: 10
Power: 1000000000000000


#### Challenges: 

##### 01. You have a floor that is 95 meters long and 58.8 meter wide. Find out the Total Area rectangular floor using python and print it. 

###### Answer: 
floor is rectangular, the formula for area = length × width.

In [4]:
# Defining dimensions of the floor
length = 95
width = 58.8

# Calculating area
area = length * width

# Printing result
print("The Total Area of the rectangular floor is:", area, "square meters")

The Total Area of the rectangular floor is: 5586.0 square meters


##### A rectangular tile is 2 meters long and 2 meters wide. 
##### 02. How many such tiles will you need to cover the floor completely?

Floor area = 95 × 58.8 = 5586.0 m²

Tile area = 2 × 2 = 4 m²

Number of tiles needed = Floor area ÷ Tile area = 5586 ÷ 4 = 1396.5

Since we can’t use half a tile in reality, we need to round up to the next whole number → 1397 tiles.

### 2. Logical Operators

#### Questions:

##### 1. What are logical operators in Python? Where do we use them?

Logical operators are used to combine conditional statements (expressions that return True or False). They help us make decisions in programs.

##### Python has three logical operators:

- and → Returns True if both conditions are true.

- or → Returns True if at least one condition is true.

- not → Returns True if the condition is false (inverts the result).

##### Where do we use them?

We use logical operators in:

- Decision making (if-else conditions)

- Loops with conditions

- Filtering data

- Validations (e.g., login checks, input checks)

##### 02. Create a Python script that defines two boolean variables, is_sunny and is_warm, with appropriate values (True or False).

In [6]:
# Defining boolean variables
is_sunny = True
is_warm = False

# Printing their values
print("Is it sunny outside?", is_sunny)
print("Is it warm outside?", is_warm)


Is it sunny outside? True
Is it warm outside? False


##### 03. Use logical operators (and, or, not) to create the following conditions:
- Check if it's both sunny and warm and print the result.
- Check if it's either sunny or warm and print the result.
- Check if it's not sunny and print the result.

In [7]:
# Defining boolean variables
is_sunny = True
is_warm = False

# 1. Check if it's both sunny and warm
print("Is it both sunny and warm?", is_sunny and is_warm)

# 2. Check if it's either sunny or warm
print("Is it either sunny or warm?", is_sunny or is_warm)

# 3. Check if it's not sunny
print("Is it not sunny?", not is_sunny)

Is it both sunny and warm? False
Is it either sunny or warm? True
Is it not sunny? False


#### Challenge: 

##### If has_umbrella = True and is_raining = False, Can you go outside without getting wet? Use logical operators to decide.

In [8]:
# Variables
has_umbrella = True
is_raining = False

# Condition using logical operators
can_go_outside = has_umbrella or not is_raining

# Print result
print("Can you go outside without getting wet?", can_go_outside)


Can you go outside without getting wet? True


### 3. Comparison Operators

#### Questions:

##### 1 .What are comparison operators? List a few real-world situations where we use them.

Comparison operators are used to compare two values. They return a boolean value (True or False) depending on the condition.

##### Here are the main ones:

- == → Equal to

- != → Not equal to

- ( > ) → Greater than

- < → Less than

- ( >= ) → Greater than or equal to

- <= → Less than or equal to

##### Real-World Situations where we use them:

###### Exams/Results:

marks = 75
print(marks >= 35)   # Did the student pass?

Checks if marks are greater than or equal to passing marks.

###### Online Shopping:

price = 1200
budget = 1000
print(price <= budget)  # Can I afford this item?

Compares price with budget.

###### Login Security:

entered_pin = 1234
correct_pin = 1234
print(entered_pin == correct_pin)  # Is the PIN correct?

Checks if the entered PIN matches the stored PIN.

###### Weather Check:

temperature = 32
print(temperature > 30)  # Is it hot outside?

Compares temperature to decide if it’s hot.

#### 02. Create a Python script that defines two variables, value1 and value2, with any numeric values.

In [10]:
# Defining two numeric variables
value1 = 25
value2 = 10

# Printing the values
print("Value 1:", value1)
print("Value 2:", value2)

Value 1: 25
Value 2: 10


#### 03. Use comparative operators (==, !=, <, >, <=, >=) to create the following conditions:
- Check if value1 is equal to value2 and print the result.
- Check if value1 is not equal to value2 and print the result.
- Check if value1 is less than value2 and print the result.
- Check if value1 is greater than value2 and print the result.
- Check if value1 is less than or equal to value2 and print the result.
- Check if value1 is greater than or equal to value2 and print the result


In [11]:
# Defining variables
value1 = 25
value2 = 10

# Comparison operations
print("Is value1 equal to value2?", value1 == value2)
print("Is value1 not equal to value2?", value1 != value2)
print("Is value1 less than value2?", value1 < value2)
print("Is value1 greater than value2?", value1 > value2)
print("Is value1 less than or equal to value2?", value1 <= value2)
print("Is value1 greater than or equal to value2?", value1 >= value2)

Is value1 equal to value2? False
Is value1 not equal to value2? True
Is value1 less than value2? False
Is value1 greater than value2? True
Is value1 less than or equal to value2? False
Is value1 greater than or equal to value2? True


#### Challenge: 
#### A person’s age is 18. Write a condition to check if the person is eligible to vote (18 or older). Print the message accordingly

In [13]:
# Age of the person
age = 18

# Checking voting eligibility
if age >= 18:
    print("You are eligible to vote.")
else:
    print("You are not eligible to vote.")

# Here, we use the comparison operator >= to check if the person is 18 or older.

You are eligible to vote.


### 4. CONCATENATION REPETITION

#### Questions:

##### 01. What is concatenation and what is repetition? Write the difference between them ?

Concatenation is joining two or more strings together using the + operator.

In [14]:
# Example:

str1 = "Hello"
str2 = "World"
result = str1 + " " + str2
print(result)  # Output: Hello World

Hello World


Repetition is repeating a string multiple times using the * operator.

In [15]:
str1 = "Hi! "
result = str1 * 3
print(result)  # Output: Hi! Hi! Hi! 

Hi! Hi! Hi! 


##### Difference Between Concatenation and Repetition in Python

##### 02. Take two variables as first_name and last_name.concatenate them.

In [21]:
# Defining variables
first_name = "Samarth"
last_name = "Gowda"

# Concatenating first name and last name
full_name = first_name + " " + last_name

# Printing the result
print("Full Name:", full_name)

Full Name: Samarth Gowda


##### 03. Print "Innomatics" 5 times in the same line.

In [20]:
# Repeating the string 5 times
print("Innomatics " * 5)

Innomatics Innomatics Innomatics Innomatics Innomatics 


#### Challenge:

#### Create a mini banner that Prints "Welcome!" followed by your full name 3 times, each on a new line.

In [22]:
# Define your full name
full_name = "Samarth Gowda"

# Print "Welcome!" followed by full name 3 times on new lines
print("Welcome!\n" + (full_name + "\n") * 3)

Welcome!
Samarth Gowda
Samarth Gowda
Samarth Gowda

